### DISCUSSION TITLE
Exit with non-zero code when smells are found

### DISCUSSION THREAD

**Discussion by lamualfa** [qltysh/qlty#1304] [Feature Requests]
## Questions

- Is `qlty smells -a` meant to fail if something goes wrong like `qlty check -a` does?
- If no, what the best practice to use `qlty smells -a`?

## What did you expect to happen?

If there's an issue on the code, `qlty smells -a` should be fail.

**Comment by brynary** [comment]
Hi @lamualfa -- Right now, `qlty smells` always exits with a success code of 0, so we would consider this to be expected behavior currently. However, I agree that it would make sense for it to behave like `qlty check` and add a `--no-fail` option to override that.

Alternatively, a bigger change would be to run the `smells` detection in the same way we run third-party plugins. That would provide the behavior to exit with a failure code automatically.
